app.rb

```diff
@@ -0,0 +1,76 @@
+require_relative 'book'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rental'
+
+def list_all_books(books)
+  if books.empty?
+    puts 'There is no book!'
+  else
+    books.each { |book| puts "Title: '#{book.title}', Author: #{book.author}" }
+  end
+end
+
+def list_all_people(people)
+  if people.empty?
+    puts 'there is no person'
+  else
+    people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+end
+
+def create_person(person)
+  puts 'do you want to create a student(1) or a teacher (2):'
+  selected_person = gets.chomp.to_i
+  print 'age:'
+  age = gets.chomp.to_i
+  print 'name:'
+  name = gets.chomp.to_s
+  case selected_person
+  when 1
+    print 'Has parent permission? [Y/N]: '
+    provided_permission = gets.chomp.capitalize
+    student_permission = true if provided_permission == 'Y'
+    student_permission = false if provided_permission == 'N'
+    person.push(Student.new(nil, age, name, parent_permission: student_permission))
+  when 2
+    print 'Specialization: '
+    specialization = gets.chomp.to_s
+    person.push(Teacher.new(age, name, specialization))
+  end
+  puts 'Person created successfully.'
+end
+
+def create_book(books)
+  print 'title:'
+  title = gets.chomp
+  print 'author:'
+  author = gets.chomp
+  books.push(Book.new(title, author))
+  puts 'the book is created successfully'
+end
+
+def create_rental(books, people, rentals)
+  puts 'Select a book from the following list by number'
+  books.map.with_index { |book, index| puts "#{index}) Title: '#{book.title}', Author: #{book.author}" }
+  selected_book = gets.chomp.to_i
+
+  puts 'Select a person from the following list by number (Not ID): '
+  people.map.with_index do |person, index|
+    puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  end
+
+  selected_person = gets.chomp.to_i
+  print 'date:'
+  provided_date = gets.chomp
+  rentals.push(Rental.new(provided_date, people[selected_person], books[selected_book]))
+  puts ' rental created succesfully'
+end
+
+def list_all_rentals_by_id(rentals)
```

Comment:
- [ ] [OPTIONAL] When trying to list rentals by id the app breaks, kindly consider fixing this so all the functionality on your app is working properly

![Screenshot_20220825_185402](https://user-images.githubusercontent.com/25227022/186735850-f8bfac33-1768-4c84-b554-98a73bda6ae5.png)

Reply:
thank you so much @Ismailco  am going to work on this optional suggestion